src/rules/require-source-directory.ts

```diff
@@ -0,0 +1,19 @@
+import { buildRule } from './helpers';
+import { RuleName } from './types';
+
+export default buildRule({
+  name: RuleName.RequireSourceDirectory,
+  description: 'Does the `src/` directory exist?',
+  dependencies: [],
+  execute: async ({ project, pass, fail }) => {
+    const entryPath = 'src';
+    const stats = await project.fs.getEntryStats(entryPath);
+    const passed = stats?.isDirectory();
+    const message = stats
+      ? `\`${entryPath}\` is not a directory when it should be.`
```

Comment:
Nit: At this point we don't know that it's a file, that's tracked by the `passed` variable. I see that this is only used when `passed` is `false`, but maybe we could refactor this to make these more obviously connected?

Reply:
Updated here. Is this clearer? [bc3f58e](https://github.com/MetaMask/module-lint/pull/33/commits/bc3f58e12c001be24380104f9e83fd78f38d1780)